Translation for "Responding to Events" (#861)
reactjs/ru.react.dev: src/content/learn/responding-to-events.md

---
title: Обработка событий
---

<Intro>

React позволяет добавлять *обработчики событий* прямо в JSX. Обработчики событий - это функции, которые вызываются в ответ на определенное событие, например, клик на элемент, наведение курсора, фокус поля формы и т.д.

</Intro>

<YouWillLearn>

* Различные способы написания обработчика событий
* Как передавать логику обработки событий от родительского компонента
* Как события распространяются и как их остановить

</YouWillLearn>

## Добавление обработчиков событий {/*adding-event-handlers*/}

> Чтобы добавить обработчик событий, сначала нужно определить функцию, а затем передать ее [в качестве пропа](/learn/passing-props-to-a-component) в соответствующий тег JSX. Например, вот кнопка, которая пока не имеет обработчика событий:

Reviewer: ```suggestion
Чтобы добавить обработчик событий, сначала нужно определить функцию, а затем передать ее [в качестве пропа](/learn/passing-props-to-a-component) в соответствующий JSX-тег. Например, вот кнопка, у которой пока нет обработчика событий:
```

Reviewer: Так, кажется, естественнее звучит

<Sandpack>

```js
export default function Button() {
  return (
    <button>

>       У меня нет обработчика событий

Reviewer: ```suggestion
      Я ничего не делаю
```

или даже 

```suggestion
      Я валяю дурака
```

Почему так сильно смысл оригинала изменил?

Author: Я ничего не делаю - кажется странным для кнопки

    </button>
  );
}
```

</Sandpack>

> Для того, чтобы при нажатии на кнопку появилось сообщение, следуйте этим шагам:

Reviewer: ```suggestion
Для того, чтобы при нажатии на кнопку появилось сообщение, выполните эти шаги:
```

Тут совершенная форма глагола нужна, потому что "появиться" тоже совершенная

1. Определите функцию `handleClick` *внутри* компонента `Button`.

> 2. Внутри функции `handleClick` реализуйте логику, которая будет вызываться при нажатии на кнопку (например, используйте функцию `alert` для отображения сообщения).

Reviewer: ```suggestion
2. Внутри функции `handleClick` реализуйте логику, которая будет срабатывать при нажатии на кнопку (используйте функцию `alert` для отображения сообщения).
```

В оригинале нет "например", к тому же это список действий, выполнив который, у меня должно появиться сообщение на экране, поэтому никакой другой функцией кроме alert это не реализовать

</Sandpack>

Вы определили функцию `handleClick` и затем [передали ее в качестве пропа](/learn/passing-props-to-a-component) в `<button>`. `handleClick` - это обработчик событий. Функции обработчиков событий:

* Обычно определяются *внутри* ваших компонентов.
* начинаются со слова `handle`, за которым следует имя события.

> Обработчики событий принято называть со слова `handle`, за которым следует имя события. Вы часто будете видеть `onClick={handleClick}`, `onMouseEnter={handleMouseEnter}` и т.д.

Reviewer: ```suggestion
Обработчики событий принято называть со слова `handle`, за которым следует имя события. Вы часто будете встречать `onClick={handleClick}`, `onMouseEnter={handleMouseEnter}` и т.п.
```

> Также можно определить обработчик события внутри JSX:

Reviewer: ```suggestion
Ещё обработчик события можно определить  внутри JSX:
```

Reviewer: Такой порядок слов, кажется, чуть естественнее

```jsx
<button onClick={function handleClick() {
  alert('Вы нажали на меня!');
}}>
```

> Или, более кратко, с помощью стрелочной функции:

Reviewer: ```suggestion
Или короче с помощью стрелочной функции:
```

Reviewer: Не уверен, что тут запятые нужны

</Sandpack>

Здесь компонент `Toolbar` рендерит компоненты `PlayButton` и `UploadButton`:

> - `PlayButton` передает `handlePlayClick` в качестве пропа `onClick` внутренней кнопке.

Reviewer 2: пожалуйста проверьте чтобы были одинаковые переносы строк.
В будущем будут приходить правки, которые нужно будет ребейзить
